env.d/60-flatpak

```diff
@@ -1,5 +1,14 @@
 #!/bin/bash
 
+# Only include the ~/.local/share/flatpak path if this is being run as a *user*
+# environment generator. If run as a *system* environment generator, only
+# include the system installations.
+if [[ "$(basename $(dirname $0))" == "user-environment-generators" ]]; then
```

Comment:
Can't the basename be changed at config time by `--with-systemduserenvgendir`?

Reply:
Hmm, that’s true, although more likely than not the distro would choose something which still ends in `user-environment-generators`, but with a different prefix.

If anyone can think of a better way to differentiate between the system and user generators from their process context, I’m all ears! I was trying to avoid installing and maintaining two almost-identical scripts.

Come to think of it, one of the installed ones could be a copy to the other, and then perhaps the file type of `$0` could be used to work out which one’s which? That’s starting to feel very hacky and implicit though.

I’ll wait for some feedback from a flatpak maintainer before making any further changes here.